packages/slate/src/models/node.js

```diff
@@ -1967,11 +1967,12 @@ class Node {
    * Validate the node against a `schema`.
    *
    * @param {Schema} schema
+   * @param {Value} value
    * @return {Function|Null}
    */
 
-  validate(schema) {
-    return schema.validateNode(this)
+  validate(schema, value) {
```

Comment:
Passing the value here defeat the memoization done on `node.validate(schema)`, because the value always change, you can never be sure if a node is still valid for the same schema.


Currently, `validate` says whether a node (and its children) is valid against a schema or not, which is limited and reasonable. I think it should stay that way. Maybe the `schema` should have a way to tell which node in particular will need to inspect the whole value.

Normalizing a node would become:

```
const requiresValue = schema.requiresValue(node);
if (requiresValue) {
  node.validate(schema, value);
} else {
  node.validate(schema);
}
```

Reply:
Thanks @Soreine , :) 

Ian gave me very similar feedback a while ago and I totally forgot to clean up this PR. I greatly appreciate the well thought out feedback, and I hope I didn't waste your time 😆